File: app/routers/favorites.py
from fastapi import APIRouter, Depends, HTTPException, status 
from schemas import FavoriteOut
from models import User, UserRole, Favorite
from sqlalchemy.orm import Session, joinedload
from app.deps import get_current_user, get_db

router = APIRouter("/favorites", tags=["favorites"])

@router.post("/{professional_user_id}", status_code=201)
def add_favorite():
    pass 

@router.get("", response_model=list[FavoriteOut])
def list_favorites():
    pass 

@router.delete("/{professional_user_id}", status_code=204)
def remove_favorite():
    pass 

@router.get("/{professional_user_id}/check")
def check_favorite():
    pass 

File: Frontend/src/api/favorites.ts
import client from "@/api/client"
import type{ Favorite } from "@/types"

export const getFavorites = async (): Promise<Favorite[]> => {
  const response = await client.get<Favorite[]>("/favorites")
  return response.data
}

export const addFavorite = async (professionalUserId: number): Promise<void> => {
  await client.post(`/favorites/${professionalUserId}`)
}

export const removeFavorite = async (
  professionalUserId: number
): Promise<void> => {
  await client.delete(`/favorites/${professionalUserId}`)
}

export const checkFavorite = async (
  professionalUserId: number
): Promise<{ is_favorited: boolean }> => {
  const response = await client.get<{ is_favorited: boolean }>(
    `/favorites/${professionalUserId}/check`
  )
  return response.data
}